README
tzdata-bb
=========

Made Babyish Timezone Available for Linux

Follow the instruction of the website listed below for resetting the timezone:

  http://www.christopherirish.com/2012/03/21/how-to-set-the-timezone-on-ubuntu-server/

This tzdata is up-to-date to tzdata-2014h


Time Zone Database
==================

For more information on the time zone data, please visit http://www.iana.org/time-zones/


Original README in tz-2014h
-------------------------------

README for the tz distribution

"What time is it?" -- Richard Deacon as The King
"Any time you want it to be." -- Frank Baxter as The Scientist
					(from the Bell System film "About Time")

The Time Zone Database (often called tz or zoneinfo) contains code and
data that represent the history of local time for many representative
locations around the globe.  It is updated periodically to reflect
changes made by political bodies to time zone boundaries, UTC offsets,
and daylight-saving rules.

Unless otherwise specified, all files in the tz code and data are in
the public domain, so clarified as of 2009-05-17 by Arthur David Olson.
The few exceptions are code derived from BSD, which uses the BSD license.

Here is a recipe for acquiring, building, installing, and testing the
tz distribution on a GNU/Linux or similar host.

	mkdir tz
	cd tz
	wget --retr-symlinks 'ftp://ftp.iana.org/tz/tz*-latest.tar.gz'
	gzip -dc tzcode-latest.tar.gz | tar -xf -
	gzip -dc tzdata-latest.tar.gz | tar -xf -

Be sure to read the comments in "Makefile" and make any changes needed
to make things right for your system, especially if you are using some
platform other than GNU/Linux.  Then run the following commands,
substituting your desired installation directory for "$HOME/tzdir":

	make TOPDIR=$HOME/tzdir install
	$HOME/tzdir/etc/zdump -v America/Los_Angeles

Historical local time information has been included here to:

*	provide a compendium of data about the history of civil time
	that is useful even if not 100% accurate;

*	give an idea of the variety of local time rules that have
	existed in the past and thus an idea of the variety that may be
	expected in the future;

*	provide a test of the generality of the local time rule description
	system.

The information in the time zone data files is by no means authoritative;
fixes and enhancements are welcome.  Please see the file CONTRIBUTING
for details.

Thanks to these Time Zone Caballeros who've made major contributions to the
time conversion package: Keith Bostic; Bob Devine; Paul Eggert; Robert Elz;
Guy Harris; Mark Horton; John Mackin; and Bradley White.  Thanks also to
Michael Bloom, Art Neilson, Stephen Prince, John Sovereign, and Frank Wales
for testing work, and to Gwillim Law for checking local mean time data.
Thanks in particular to Arthur David Olson, the project's founder and first
maintainer, to whom the time zone community owes the greatest debt of all.
None of them are responsible for remaining errors.

Look in <ftp://ftp.iana.org/tz/releases/> for updated versions of these files.

Please send comments or information to tz@iana.org.


NEWS
News for the tz database

<<<<<<< HEAD
=======
Unreleased, experimental changes

  Changes affecting future time stamps

    A new Zone Pacific/Bougainville, for the part of Papua New Guinea
    that plans to switch from UTC+10 to UTC+11 on 2014-12-28 at 02:00.
    (Thanks to Kiley Walbom for the heads-up.)

    Since Belarus is not changing its clocks even though Moscow is,
    the time zone abbreviation in Europe/Minsk is changing from FET
    to its more-traditional value MSK on 2014-10-26 at 01:00.
    (Thanks to Alexander Bokovoy for the heads-up about Belarus.)

  Changes affecting past time stamps

    Many time stamps have been corrected for Asia/Ho_Chi_Minh before 1976
    (thanks to Trần Ngọc Quân for an indirect pointer to Trần Tiến Bình's
    authoritative book).  Asia/Ho_Chi_Minh has been added to
    zone1970.tab, to give tzselect users in Vietnam two choices,
    since north and south Vietnam disagreed after our 1970 cutoff.
    The new abbreviation IDT stands for the pre-1976 use of UT+8 in
    Indochina, to distinguish it better from ICT (UT+7).

    Asia/Phnom_Penh and Asia/Vientiane have been turned into links, as
    they differed from existing zones only for older time stamps.  As
    usual, these changes affect pre-1970 time stamps only.  Their old
    contents have been moved to the 'backzone' file.

  Changes affecting code

    If USG_COMPAT is defined and the requested time stamp is standard time,
    the tz library's localtime and mktime functions now set the extern
    variable timezone to a value appropriate for that time stamp; and
    similarly for ALTZONE, daylight saving time, and the altzone variable.
    This change is a companion to the tzname change in 2014h, and is
    designed to make timezone and altzone more compatible with tzname.

    The tz library's functions now set errno to EOVERFLOW if they fail
    because the result cannot be represented.  ctime and ctime_r now
    return NULL and set errno when a time stamp is out of range, rather
    than having undefined behavior.

    Some bugs associated with the new 2014g functions have been fixed.
    This includes a bug that largely incapacitated the new functions
    time2posix_z and posix2time_z.  (Thanks to Christos Zoulas.)
    It also includes some uses of uninitialized variables after tzalloc.
    The new code uses the standard type 'ssize_t', which the Makefile
    now gives porting advice about.

>>>>>>> grandpa/master
Release 2014h - 2014-09-25 18:59:03 -0700

  Changes affecting past time stamps

    America/Jamaica's 1974 spring-forward transition was Jan. 6, not Apr. 28.

    Shanks says Asia/Novokuznetsk switched from LMT (not "NMT") on 1924-05-01,
    not 1920-01-06.  The old entry was based on a misinterpretation of Shanks.

    Some more zones have been turned into links, when they differed
    from existing zones only for older time stamps.  As usual,
    these changes affect UTC offsets in pre-1970 time stamps only.
    Their old contents have been moved to the 'backzone' file.
    The affected zones are: Africa/Blantyre, Africa/Bujumbura,
    Africa/Gaborone, Africa/Harare, Africa/Kigali, Africa/Lubumbashi,
    Africa/Lusaka, Africa/Maseru, and Africa/Mbabane.

  Changes affecting code

    zdump -V and -v now output gmtoff= values on all platforms,
    not merely on platforms defining TM_GMTOFF.

    The tz library's localtime and mktime functions now set tzname to a value
    appropriate for the requested time stamp, and zdump now uses this
    on platforms not defining TM_ZONE, fixing a 2014g regression.
    (Thanks to Tim Parenti for reporting the problem.)

    The tz library no longer sets tzname if localtime or mktime fails.

    zdump -c no longer mishandles transitions near year boundaries.
    (Thanks to Tim Parenti for reporting the problem.)

    An access to uninitalized data has been fixed.
    (Thanks to Jörg Richter for reporting the problem.)

    When THREAD_SAFE is defined, the code ports to the C11 memory model.
    A memory leak has been fixed if ALL_STATE and THREAD_SAFE are defined
    and two threads race to initialize data used by gmtime-like functions.
    (Thanks to Andy Heninger for reporting the problems.)

  Changes affecting build procedure

    'make check' now checks better for properly-sorted data.

  Changes affecting documentation and commentary

    zdump's gmtoff=N output is now documented, and its isdst=D output
    is now documented to possibly output D values other than 0 or 1.

    zdump -c's treatment of years is now documented to use the
    Gregorian calendar and Universal Time without leap seconds,
    and its behavior at cutoff boundaries is now documented better.
    (Thanks to Arthur David Olson and Tim Parenti for reporting the problems.)

    Programs are now documented to use the proleptic Gregorian calendar.
    (Thanks to Alan Barrett for the suggestion.)

    Fractional-second GMT offsets have been documented for civil time
    in 19th-century Chennai, Jakarta, and New York.


Release 2014g - 2014-08-28 12:31:23 -0700

  Changes affecting future time stamps

    Turks & Caicos is switching from US eastern time to UTC-4 year-round,
    modeled as a switch from EST/EDT to AST on 2014-11-02 at 02:00.

  Changes affecting past time stamps

    Time in Russia or the USSR before 1926 or so has been corrected by
    a few seconds in the following zones: Asia/Irkutsk,
    Asia/Krasnoyarsk, Asia/Omsk, Asia/Samarkand, Asia/Tbilisi,
    Asia/Vladivostok, Asia/Yakutsk, Europe/Riga, Europe/Samara.  For
    Asia/Yekaterinburg the correction is a few minutes.  (Thanks to
    Vladimir Karpinsky.)

<<<<<<< HEAD

private.h
#ifndef PRIVATE_H

#define PRIVATE_H

/*
<<<<<<< HEAD
** This header is for use ONLY with the time zone code.
** There is no guarantee that it will remain unchanged,
** or even that it will remain at all.
=======
** This file is in the public domain, so clarified as of
** 1996-06-05 by Arthur David Olson.
*/

/*
** This header is for use ONLY with the time conversion code.
** There is no guarantee that it will remain unchanged,
** or that it will remain at all.
>>>>>>> grandpa/master
** Do NOT copy it to any system include directory.
** Thank you!
*/

#define GRANDPARENTED	"Local time zone must be set--see zic manual page"

/*
** Defaults for preprocessor symbols.
** You can override these in your C compiler options, e.g. '-DHAVE_ADJTIME=0'.
*/

<<<<<<< HEAD
#ifndef const

#ifndef OK_CONST
#ifndef NO_CONST
#ifdef __STDC__
#define OK_CONST
#endif /* defined __STDC__ */
#endif /* !defined NO_CONST */
#endif /* !defined OK_CONST */

#ifndef OK_CONST
#define const
#endif /* !defined OK_CONST */

#endif /* !defined const */

/*
** void
*/

#ifndef void

#ifndef OK_VOID
#ifndef NO_VOID
#ifdef __STDC__
#define OK_VOID
#endif /* defined __STDC__ */
#ifdef sun
#define OK_VOID
#endif /* defined sun */
#ifdef vax
#define OK_VOID
#endif /* defined vax */
#endif /* !defined NO_VOID */
#endif /* !defined OK_VOID */

#ifndef OK_VOID
#define void char
#endif /* !defined OK_VOID */

#endif /* !defined void */

/*
** P((args))
*/

#ifndef P

#ifndef OK_PROTO
#ifndef NO_PROTO
#ifdef __STDC__
#define OK_PROTO
#endif /* defined __STDC__ */
#endif /* !defined NO_PROTO */
#endif /* !defined OK_PROTO */

#ifdef OK_PROTO
#define P(x)	x
#endif /* defined OK_PROTO */

#ifndef OK_PROTO
#define ASTERISK	*
#define P(x)	( /ASTERISK x ASTERISK/ )
#endif /* !defined OK_PROTO */

#endif /* !defined P */

/*
** Includes.  First the ones we expect to show up on all systems.
*/

/*
** <assert.h>
*/

#ifdef ASSERT_H
#include ASSERT_H
#endif /* defined ASSERT_H */

#ifndef ASSERT_H
#include "assert.h"
#endif /* !defined ASSERT_H */

/*
** <ctype.h>
*/

#ifdef CTYPE_H
#include CTYPE_H
#endif /* defined CTYPE_H */

#ifndef CTYPE_H
#include "ctype.h"
#endif /* !defined CTYPE_H */

/*
** <errno.h>
*/

#ifdef ERRNO_H
#include ERRNO_H
#endif /* defined ERRNO_H */

#ifndef ERRNO_H
#include "errno.h"
#endif /* !defined ERRNO_H */

/*
** <stdio.h>
*/

#ifdef STDIO_H
#include STDIO_H
#endif /* defined STDIO_H */

#ifndef STDIO_H
#include "stdio.h"
extern int	fclose P((FILE * stream));
extern int	fflush P((FILE * stream));
extern int	fprintf P((FILE * stream, const char * format, ...));
extern int	fputc P((int c, FILE * stream));
extern int	fputs P((const char * s, FILE * stream));
extern int	fread P((char * p, int s, int n, FILE * stream));
extern int	fscanf P((FILE * stream, const char * format, ...));
extern int	fseek P((FILE * stream, long int offset, int whence));
extern int	fwrite P((char * p, int s, int n, FILE * stream));
extern int	printf P((const char * format, ...));
extern int	scanf P((const char * format, ...));
extern int	sscanf P((const char * string, const char * format, ...));
extern int	ungetc P((int c, FILE * stream));
/*
** To prevent gcc squawks when in "prototypes required" mode. . .
*/
extern int	_filbuf P((FILE * stream));
extern int	_flsbuf P((unsigned char c, FILE * stream));

#ifndef remove
extern int	unlink P((const char * filename));
#define remove	unlink
#endif /* !defined remove */

#ifndef FILENAME_MAX

#ifndef MAXPATHLEN
#ifdef unix
#include <sys/param.h>
#endif /* defined unix */
#endif /* !defined MAXPATHLEN */

#ifdef MAXPATHLEN
#define FILENAME_MAX	MAXPATHLEN
#else /* !defined MAXPATHLEN */
#define FILENAME_MAX	1024		/* Pure guesswork */
#endif /* !defined MAXPATHLEN */

#endif /* !defined FILENAME_MAX */

#ifdef USG
extern void	rewind P((FILE * stream));
extern int	sprintf P((char * string, const char * format, ...));
#endif /* defined USG */
#ifndef USG
extern int	rewind P((FILE * stream));
extern char *	sprintf P((char * string, const char * format, ...));
#endif /* !defined USG */

#endif /* !defined STDIO_H */
=======
#ifndef HAVE_ADJTIME
#define HAVE_ADJTIME		1
#endif /* !defined HAVE_ADJTIME */

#ifndef HAVE_GETTEXT
#define HAVE_GETTEXT		0
#endif /* !defined HAVE_GETTEXT */

#ifndef HAVE_INCOMPATIBLE_CTIME_R
#define HAVE_INCOMPATIBLE_CTIME_R	0
#endif /* !defined INCOMPATIBLE_CTIME_R */

#ifndef HAVE_LINK
#define HAVE_LINK		1
#endif /* !defined HAVE_LINK */

#ifndef HAVE_SETTIMEOFDAY
#define HAVE_SETTIMEOFDAY	3
#endif /* !defined HAVE_SETTIMEOFDAY */

#ifndef HAVE_SYMLINK
#define HAVE_SYMLINK		1
#endif /* !defined HAVE_SYMLINK */

#ifndef HAVE_SYS_STAT_H
#define HAVE_SYS_STAT_H		1
#endif /* !defined HAVE_SYS_STAT_H */

#ifndef HAVE_SYS_WAIT_H
#define HAVE_SYS_WAIT_H		1
#endif /* !defined HAVE_SYS_WAIT_H */

#ifndef HAVE_UNISTD_H
#define HAVE_UNISTD_H		1
#endif /* !defined HAVE_UNISTD_H */

#ifndef HAVE_UTMPX_H
#define HAVE_UTMPX_H		0
#endif /* !defined HAVE_UTMPX_H */

#ifndef NETBSD_INSPIRED
# define NETBSD_INSPIRED 1
#endif

#if HAVE_INCOMPATIBLE_CTIME_R
#define asctime_r _incompatible_asctime_r
#define ctime_r _incompatible_ctime_r
#endif /* HAVE_INCOMPATIBLE_CTIME_R */

/* Enable tm_gmtoff and tm_zone on GNUish systems.  */
#define _GNU_SOURCE 1
/* Fix asctime_r on Solaris 10.  */
#define _POSIX_PTHREAD_SEMANTICS 1
/* Enable strtoimax on Solaris 10.  */
#define __EXTENSIONS__ 1

/*
** Nested includes
*/

/* Avoid clashes with NetBSD by renaming NetBSD's declarations.  */
#define localtime_rz sys_localtime_rz
#define mktime_z sys_mktime_z
#define posix2time_z sys_posix2time_z
#define time2posix_z sys_time2posix_z
#define timezone_t sys_timezone_t
#define tzalloc sys_tzalloc
#define tzfree sys_tzfree
#include <time.h>
#undef localtime_rz
#undef mktime_z
#undef posix2time_z
#undef time2posix_z
#undef timezone_t
#undef tzalloc
#undef tzfree

#include "sys/types.h"	/* for time_t */
#include "stdio.h"
#include "string.h"
#include "limits.h"	/* for CHAR_BIT et al. */
#include "stdlib.h"

#include "errno.h"

#ifndef EOVERFLOW
# define EOVERFLOW EINVAL
#endif

#if HAVE_GETTEXT
#include "libintl.h"
#endif /* HAVE_GETTEXT */

#if HAVE_SYS_WAIT_H
#include <sys/wait.h>	/* for WIFEXITED and WEXITSTATUS */
#endif /* HAVE_SYS_WAIT_H */

#ifndef WIFEXITED
#define WIFEXITED(status)	(((status) & 0xff) == 0)
#endif /* !defined WIFEXITED */
#ifndef WEXITSTATUS
#define WEXITSTATUS(status)	(((status) >> 8) & 0xff)
#endif /* !defined WEXITSTATUS */

#if HAVE_UNISTD_H
#include "unistd.h"	/* for F_OK, R_OK, and other POSIX goodness */
#endif /* HAVE_UNISTD_H */

#ifndef HAVE_STRFTIME_L
# if _POSIX_VERSION < 200809
#  define HAVE_STRFTIME_L 0
# else
#  define HAVE_STRFTIME_L 1
# endif
#endif

#ifndef F_OK
#define F_OK	0
#endif /* !defined F_OK */
#ifndef R_OK
#define R_OK	4
#endif /* !defined R_OK */

/* Unlike <ctype.h>'s isdigit, this also works if c < 0 | c > UCHAR_MAX. */
#define is_digit(c) ((unsigned)(c) - '0' <= 9)

/*
** Define HAVE_STDINT_H's default value here, rather than at the
** start, since __GLIBC__'s value depends on previously-included
** files.
** (glibc 2.1 and later have stdint.h, even with pre-C99 compilers.)
*/
#ifndef HAVE_STDINT_H
#define HAVE_STDINT_H \
   (199901 <= __STDC_VERSION__ \
    || 2 < __GLIBC__ + (1 <= __GLIBC_MINOR__)	\
    || __CYGWIN__)
#endif /* !defined HAVE_STDINT_H */

#if HAVE_STDINT_H
#include "stdint.h"
#endif /* !HAVE_STDINT_H */

#ifndef HAVE_INTTYPES_H
# define HAVE_INTTYPES_H HAVE_STDINT_H
#endif
#if HAVE_INTTYPES_H
# include <inttypes.h>
#endif

/* Pre-C99 GCC compilers define __LONG_LONG_MAX__ instead of LLONG_MAX.  */
#ifdef __LONG_LONG_MAX__
# ifndef LLONG_MAX
#  define LLONG_MAX __LONG_LONG_MAX__
# endif
# ifndef LLONG_MIN
#  define LLONG_MIN (-1 - LLONG_MAX)
# endif
#endif

#ifndef INT_FAST64_MAX
# ifdef LLONG_MAX
typedef long long	int_fast64_t;
#  define INT_FAST64_MIN LLONG_MIN
#  define INT_FAST64_MAX LLONG_MAX
# else
#  if (LONG_MAX >> 31) < 0xffffffff
Please use a compiler that supports a 64-bit integer type (or wider);
you may need to compile with "-DHAVE_STDINT_H".
#  endif
typedef long		int_fast64_t;
#  define INT_FAST64_MIN LONG_MIN
#  define INT_FAST64_MAX LONG_MAX
# endif
#endif

#ifndef SCNdFAST64
# if INT_FAST64_MAX == LLONG_MAX
#  define SCNdFAST64 "lld"
# else
#  define SCNdFAST64 "ld"
# endif
#endif

#ifndef INT_FAST32_MAX
# if INT_MAX >> 31 == 0
typedef long int_fast32_t;
#  define INT_FAST32_MAX LONG_MAX
#  define INT_FAST32_MIN LONG_MIN
# else
typedef int int_fast32_t;
#  define INT_FAST32_MAX INT_MAX
#  define INT_FAST32_MIN INT_MIN
# endif
#endif

#ifndef INTMAX_MAX
# ifdef LLONG_MAX
typedef long long intmax_t;
#  define strtoimax strtoll
#  define INTMAX_MAX LLONG_MAX
#  define INTMAX_MIN LLONG_MIN
# else
typedef long intmax_t;
#  define strtoimax strtol
#  define INTMAX_MAX LONG_MAX
#  define INTMAX_MIN LONG_MIN
# endif
#endif

#ifndef PRIdMAX
# if INTMAX_MAX == LLONG_MAX
#  define PRIdMAX "lld"
# else
#  define PRIdMAX "ld"
# endif
#endif

#ifndef UINTMAX_MAX
# if defined ULLONG_MAX || defined __LONG_LONG_MAX__
typedef unsigned long long uintmax_t;
# else
typedef unsigned long uintmax_t;
# endif
#endif

#ifndef PRIuMAX
# if defined ULLONG_MAX || defined __LONG_LONG_MAX__
#  define PRIuMAX "llu"
# else
#  define PRIuMAX "lu"
# endif
#endif

#ifndef INT32_MAX
#define INT32_MAX 0x7fffffff
#endif /* !defined INT32_MAX */
#ifndef INT32_MIN
#define INT32_MIN (-1 - INT32_MAX)
#endif /* !defined INT32_MIN */

#ifndef SIZE_MAX
#define SIZE_MAX ((size_t) -1)
#endif

#if 2 < __GNUC__ + (96 <= __GNUC_MINOR__)
# define ATTRIBUTE_CONST __attribute__ ((const))
# define ATTRIBUTE_PURE __attribute__ ((__pure__))
# define ATTRIBUTE_FORMAT(spec) __attribute__ ((__format__ spec))
#else
# define ATTRIBUTE_CONST /* empty */
# define ATTRIBUTE_PURE /* empty */
# define ATTRIBUTE_FORMAT(spec) /* empty */
#endif

#if !defined _Noreturn && __STDC_VERSION__ < 201112
# if 2 < __GNUC__ + (8 <= __GNUC_MINOR__)
#  define _Noreturn __attribute__ ((__noreturn__))
# else
#  define _Noreturn
# endif
#endif

#if __STDC_VERSION__ < 199901 && !defined restrict
# define restrict /* empty */
#endif

/*
** Workarounds for compilers/systems.
*/

/*
** Compile with -Dtime_tz=T to build the tz package with a private
** time_t type equivalent to T rather than the system-supplied time_t.
** This debugging feature can test unusual design decisions
** (e.g., time_t wider than 'long', or unsigned time_t) even on
** typical platforms.
*/
#ifdef time_tz
# ifdef LOCALTIME_IMPLEMENTATION
static time_t sys_time(time_t *x) { return time(x); }
# endif

# undef  ctime
# define ctime tz_ctime
# undef  ctime_r
# define ctime_r tz_ctime_r
# undef  difftime
# define difftime tz_difftime
# undef  gmtime
# define gmtime tz_gmtime
# undef  gmtime_r
# define gmtime_r tz_gmtime_r
# undef  localtime
# define localtime tz_localtime
# undef  localtime_r
# define localtime_r tz_localtime_r
# undef  mktime
# define mktime tz_mktime
# undef  offtime
# define offtime tz_offtime
# undef  posix2time
# define posix2time tz_posix2time
# undef  time
# define time tz_time
# undef  time2posix
# define time2posix tz_time2posix
# undef  time_t
# define time_t tz_time_t
# undef  timegm
# define timegm tz_timegm
# undef  timelocal
# define timelocal tz_timelocal
# undef  timeoff
# define timeoff tz_timeoff

typedef time_tz time_t;

char *ctime(time_t const *);
char *ctime_r(time_t const *, char *);
double difftime(time_t, time_t);
struct tm *gmtime(time_t const *);
struct tm *gmtime_r(time_t const *restrict, struct tm *restrict);
struct tm *localtime(time_t const *);
struct tm *localtime_r(time_t const *restrict, struct tm *restrict);
time_t mktime(struct tm *);
time_t time(time_t *);
#endif

/*
** Some time.h implementations don't declare asctime_r.
** Others might define it as a macro.
** Fix the former without affecting the latter.
*/

#ifndef asctime_r
extern char *	asctime_r(struct tm const *restrict, char *restrict);
#endif

/*
** The STD_INSPIRED functions are similar, but most also need
** declarations if time_tz is defined.
*/

#ifdef STD_INSPIRED
# if !defined tzsetwall
void tzsetwall(void);
# endif
# if !defined offtime || defined time_tz
struct tm *offtime(time_t const *, long);
# endif
# if !defined timegm || defined time_tz
time_t timegm(struct tm *);
# endif
# if !defined timelocal || defined time_tz
time_t timelocal(struct tm *);
# endif
# if !defined timeoff || defined time_tz
time_t timeoff(struct tm *, long);
# endif
# if !defined time2posix || defined time_tz
time_t time2posix(time_t);
# endif
# if !defined posix2time || defined time_tz
time_t posix2time(time_t);
# endif
#endif

/* Infer TM_ZONE on systems where this information is known, but suppress
   guessing if NO_TM_ZONE is defined.  Similarly for TM_GMTOFF.  */
#if (defined __GLIBC__ \
     || defined __FreeBSD__ || defined __NetBSD__ || defined __OpenBSD__ \
     || (defined __APPLE__ && defined __MACH__))
# if !defined TM_GMTOFF && !defined NO_TM_GMTOFF
#  define TM_GMTOFF tm_gmtoff
# endif
# if !defined TM_ZONE && !defined NO_TM_ZONE
#  define TM_ZONE tm_zone
# endif
#endif

/*
** Define functions that are ABI compatible with NetBSD but have
** better prototypes.  NetBSD 6.1.4 defines a pointer type timezone_t
** and labors under the misconception that 'const timezone_t' is a
** pointer to a constant.  This use of 'const' is ineffective, so it
** is not done here.  What we call 'struct state' NetBSD calls
** 'struct __state', but this is a private name so it doesn't matter.
*/
#if NETBSD_INSPIRED
typedef struct state *timezone_t;
struct tm *localtime_rz(timezone_t restrict, time_t const *restrict,
			struct tm *restrict);
time_t mktime_z(timezone_t restrict, struct tm *restrict);
timezone_t tzalloc(char const *);
void tzfree(timezone_t);
# ifdef STD_INSPIRED
#  if !defined posix2time_z || defined time_tz
time_t posix2time_z(timezone_t, time_t) ATTRIBUTE_PURE;
#  endif
#  if !defined time2posix_z || defined time_tz
time_t time2posix_z(timezone_t, time_t) ATTRIBUTE_PURE;
#  endif
# endif
#endif

/*
** Private function declarations.
*/

char *		icatalloc(char * old, const char * new);
char *		icpyalloc(const char * string);
const char *	scheck(const char * string, const char * format);

/*
** Finally, some convenience items.
*/

#if __STDC_VERSION__ < 199901
# define true 1
# define false 0
# define bool int
#else
# include <stdbool.h>
#endif

#ifndef TYPE_BIT
#define TYPE_BIT(type)	(sizeof (type) * CHAR_BIT)
#endif /* !defined TYPE_BIT */

#ifndef TYPE_SIGNED
#define TYPE_SIGNED(type) (((type) -1) < 0)
#endif /* !defined TYPE_SIGNED */

/* The minimum and maximum finite time values.  */
static time_t const time_t_min =
  (TYPE_SIGNED(time_t)
   ? (time_t) -1 << (CHAR_BIT * sizeof (time_t) - 1)
   : 0);
static time_t const time_t_max =
  (TYPE_SIGNED(time_t)
   ? - (~ 0 < 0) - ((time_t) -1 << (CHAR_BIT * sizeof (time_t) - 1))
   : -1);

#ifndef INT_STRLEN_MAXIMUM
/*
** 302 / 1000 is log10(2.0) rounded up.
** Subtract one for the sign bit if the type is signed;
** add one for integer division truncation;
** add one more for a minus sign if the type is signed.
*/
#define INT_STRLEN_MAXIMUM(type) \
	((TYPE_BIT(type) - TYPE_SIGNED(type)) * 302 / 1000 + \
	1 + TYPE_SIGNED(type))
#endif /* !defined INT_STRLEN_MAXIMUM */

/*
** INITIALIZE(x)
*/

#ifdef lint
# define INITIALIZE(x)	((x) = 0)
#else
# define INITIALIZE(x)
#endif

#ifndef UNINIT_TRAP
# define UNINIT_TRAP 0
#endif
>>>>>>> grandpa/master

/*
** For the benefit of GNU folk...
** '_(MSGID)' uses the current locale's message library string for MSGID.
** The default is to use gettext if available, and use MSGID otherwise.
*/

<<<<<<< HEAD
#ifdef STRING_H
#include STRING_H
#endif /* defined STRING_H */

#ifndef STRING_H
#include "string.h"
#ifdef sun
#include "memory.h"
#endif /* defined sun */
#endif /* !defined STRING_H */

/*
** <time.h>
*/

#ifdef TIME_H
#include TIME_H
#endif /* defined TIME_H */

#ifndef TIME_H
#include "time.h"
#endif /* !defined TIME_H */

#ifdef STDLIB_H
#include STDLIB_H
#endif /* defined STDLIB_H */

#ifndef STDLIB_H
extern int	atoi P((const char * nptr));
extern char *	getenv P((char * name));
extern char *	malloc P((unsigned size));
extern char *	calloc P((unsigned nelem, unsigned elsize));
extern char *	realloc P((void * pointer, unsigned size));
extern int	system P((const char * command));
extern int	abs P((int j));

#ifdef USG
extern void	exit P((int status));
extern void	free P((void * pointer));
extern void	qsort P((void * base, unsigned nmemb, unsigned size,
			int (*compar)(const void * left, const void * right)));
#endif /* defined USG */
#ifndef USG
extern int	exit P((int status));
extern int	free P((void * pointer));
extern int	qsort P((void * base, unsigned nmemb, unsigned size,
			int (*compar)(const void * left, const void * right)));
#endif /* !defined USG */
#endif /* !defined STDLIB_H */

#ifndef EXIT_SUCCESS
#define EXIT_SUCCESS	0
#endif /* !defined EXIT_SUCCESS */

#ifndef EXIT_FAILURE
#define EXIT_FAILURE	1
#endif /* !defined EXIT_FAILURE */

#ifdef __STDC__

typedef void *		genericptr_t;
#define alloc_size_t	size_t
#define qsort_size_t	size_t
#define fread_size_t	size_t
#define fwrite_size_t	size_t

#else /* !defined __STDC__ */

#ifndef genericptr_t
typedef char *		genericptr_t;
#endif /* !defined genericptr_t */

#ifndef alloc_size_t
typedef unsigned	alloc_size_t;
#endif /* !defined alloc_size_t */

#ifndef qsort_size_t
#ifdef unix
#include "sys/param.h"
#endif /* defined unix */
#ifdef BSD
typedef int		qsort_size_t;
#else /* !defined BSD */
typedef unsigned	qsort_size_t;
#endif /* !defined BSD */
#endif /* !defined qsort_size_t */

#ifndef fread_size_t
typedef int		fread_size_t;
#endif /* !defined fread_size_t */

#ifndef fwrite_size_t
typedef int		fwrite_size_t;
#endif /* !defined fwrite_size_t */

#endif /* !defined __STDC__ */

/*
** UNIX is a registered trademark of AT&T.
** VAX is a trademark of Digital Equipment Corporation.
*/
=======
#ifndef _
#if HAVE_GETTEXT
#define _(msgid) gettext(msgid)
#else /* !HAVE_GETTEXT */
#define _(msgid) msgid
#endif /* !HAVE_GETTEXT */
#endif /* !defined _ */

#if !defined TZ_DOMAIN && defined HAVE_GETTEXT
# define TZ_DOMAIN "tz"
#endif

#if HAVE_INCOMPATIBLE_CTIME_R
#undef asctime_r
#undef ctime_r
char *asctime_r(struct tm const *, char *);
char *ctime_r(time_t const *, char *);
#endif /* HAVE_INCOMPATIBLE_CTIME_R */

#ifndef YEARSPERREPEAT
#define YEARSPERREPEAT		400	/* years before a Gregorian repeat */
#endif /* !defined YEARSPERREPEAT */

/*
** The Gregorian year averages 365.2425 days, which is 31556952 seconds.
*/

#ifndef AVGSECSPERYEAR
#define AVGSECSPERYEAR		31556952L
#endif /* !defined AVGSECSPERYEAR */

#ifndef SECSPERREPEAT
#define SECSPERREPEAT		((int_fast64_t) YEARSPERREPEAT * (int_fast64_t) AVGSECSPERYEAR)
#endif /* !defined SECSPERREPEAT */

#ifndef SECSPERREPEAT_BITS
#define SECSPERREPEAT_BITS	34	/* ceil(log2(SECSPERREPEAT)) */
#endif /* !defined SECSPERREPEAT_BITS */
>>>>>>> grandpa/master

#endif /* !defined PRIVATE_H */
